README.md

```diff
@@ -4,3 +4,48 @@
 
 1. Download Go 1.17 from [here](https://go.dev/doc/install).
 2. Download `golangci-lint` from [here](https://golangci-lint.run/usage/install/).
+
+## Contribution Guidelines
+A lot of these guidelines are from the [SOMAS2020 repo](https://github.com/SOMAS2020/SOMAS2020/blob/main/docs/SETUP.md) :)
+### Coding Rules
+1. You're encouraged to use [VSCode](https://code.visualstudio.com/) with the [Go extension](https://code.visualstudio.com/docs/languages/go).
+2. Trust the language server. Red lines == death. Yellow lines == close to death. An example where it might be very tempting to let yellow lines pass are in `struct`s:
+```golang
+type S struct {
+    name string
+    age  int
+}
+
+s1 := S {"pitt", 42} // NO
+s2 := S {"pitt"} // NO (even though it initialises age to 0)
+
+s3 := S{name: "pitt", age: 42} // OK, if we add fields into S, this will still be correct
+s4 := S{name: "pittson"} // OK if `pittson`'s age is 0
+```
+3. Write tests where required. There is ample guide online on how to do this in Golang. Tests will be run alongside CI when you pull into the main repo. If anyone breaks anything, it's easy to observe that if you have tests. Otherwise, your code will be broken unknowingly.
+
+4. DO NOT TOUCH code you don't own unless you have a good reason to. If you have a good reason to, do it in a separate PR and notify the owners of the code.
+
+5. Do not use `panic` or `die` - return an `error` instead!
+
+6. Do not use system-specific packages (e.g. `internal/syscall/unix`).
+
+7. Use the superior `errors.Errorf` to create your errors so that we have a stack trace.
+### Code Reviews and PRs
+- Do not push to the `main` branch. Make your own branch and create a PR into `main` when it's ready for review.
+- When working on your own team's features, please name your branch as: `teamX/FEATURE_NAME/WHATEVER_YOU_LIKE_HERE`
```

Comment:
Maybe we can make this section more specific to agent design to make it clearer.

Reply:
Wouldn't this apply to the tower team too, not just the agent teams?

Reply:
I agree, but I think providing a specific example could be helpful.

Reply:
Can you give an example?

Reply:
Actually, I think that I have overthought the issue 😆. This is fine.

Reply:
Actually, I think that I have overthought the issue 😆. This is fine.